opengever/maintenance/scripts/set_external_reference_for_workspaces.py

```diff
@@ -0,0 +1,52 @@
+"""
+bin/instance run ./scripts/set_external_reference_for_workspaces.py path/to/file
+Set linked dossier oguid as external_reference for linked workspaces.
+"""
+
+from opengever.maintenance.debughelpers import setup_app
+from opengever.maintenance.debughelpers import setup_option_parser
+from opengever.maintenance.debughelpers import setup_plone
+from plone import api
+import json
+import sys
+import transaction
+
+
+SEPARATOR = '-' * 78
+
+
+def set_external_reference(file_path):
+    with open(file_path) as file_:
+        data = json.load(file_)
+    uids = data.keys()
+    query = {"object_provides": "opengever.workspace.interfaces.IWorkspace"}
+    catalog = api.portal.get_tool('portal_catalog')
+    workspace_brains = catalog.unrestrictedSearchResults(query)
+    for brain in workspace_brains:
+        workspace = brain.getObject()
+        if workspace.UID() in uids:
+            workspace.external_reference = data[workspace.UID()]
+        else:
+            workspace.external_reference = ''
```

Comment:
This should be unicode (so, `u''`). A `zope.schema.TextField` always expects unicode, and depending on how it's being set, will complain. Worse, mixed types of unicode vs. bytestrings across different objects can lead to hard to debug errors once they are getting compared to each other or concatenated.

In the PR in opengever.core this was not an issue because everything passes through JSON + the REST API, which already leads to a conversion to unicode.